src/pages/libs/storage.ts

```diff
@@ -0,0 +1,73 @@
+
+import Browser from "webextension-polyfill";
+const isUpdating: {
+  [key: string]: boolean
+} = {}
+
+const lastUpdated: {
+  [key: string]: number
+} = {}
+
+const currentData: {
+  [key: string]: any
+} = {}
+
+export async function fetchData({
+  key, updateFrequency, getData,
+}: {
+  updateFrequency: number,
+  key: string,
+  getData: any,
+}): Promise<any> {
+
+  // return current data while updating
+  if (isUpdating[key])
+    return currentData[key];
+
+  const timeNow = Math.floor(Date.now() / 1000);
+  const lastUpdatedTime = lastUpdated[key] ?? 0;
+
+  // return current data if it's not time to update
+  if (timeNow - lastUpdatedTime < updateFrequency)
+    return currentData[key]
+
+
+  console.log("Updating storage data for", key)
+
+  isUpdating[key] = true;
+  currentData[key] = fetchData()
```

Comment:
why pick the same name for both functions? i imagine theres no recursion here but why not just pick different names since theres no cost and eliminate any possible issues?

Reply:
fixed it